daemon/algod/api/server/v2/handlers.go

```diff
@@ -0,0 +1,449 @@
+				Amount:   holding.Amount,
+				AssetId:  uint64(curid),
+				Creator:  creator,
+				IsFrozen: holding.Frozen,
+			}
+
+			assets = append(assets, holding)
+		}
+	}
+
+	createdAssets := make([]generated.Asset, 0)
+	if len(record.AssetParams) > 0 {
+		for idx, params := range record.AssetParams {
+			assetParams := generated.AssetParams{
+				Creator:       address,
+				Total:         params.Total,
+				Decimals:      uint64(params.Decimals),
+				DefaultFrozen: &params.DefaultFrozen,
+				MetadataHash:  byteOrNil(params.MetadataHash[:]),
+				Name:          strOrNil(params.AssetName),
+				UnitName:      strOrNil(params.UnitName),
+				Url:           strOrNil(params.URL),
+				Clawback:      addrOrNil(params.Clawback),
+				Freeze:        addrOrNil(params.Freeze),
+				Manager:       addrOrNil(params.Manager),
+				Reserve:       addrOrNil(params.Reserve),
+			}
+			asset := generated.Asset{
+				Index:  uint64(idx),
+				Params: assetParams,
+			}
+			createdAssets = append(createdAssets, asset)
+		}
+	}
+
+	var apiParticipation *generated.AccountParticipation
+	if record.VoteID != (crypto.OneTimeSignatureVerifier{}) {
+		apiParticipation = &generated.AccountParticipation{
+			VoteParticipationKey:      byteOrNil(record.VoteID[:]),
+			SelectionParticipationKey: byteOrNil(record.SelectionID[:]),
+			VoteFirstValid:            numOrNil(uint64(record.VoteFirstValid)),
+			VoteLastValid:             numOrNil(uint64(record.VoteLastValid)),
+			VoteKeyDilution:           numOrNil(uint64(record.VoteKeyDilution)),
+		}
+	}
+
+	response := generated.AccountResponse{
+		Type:                        nil,
+		Round:                       uint64(lastRound),
+		Address:                     addr.String(),
+		Amount:                      amount.Raw,
+		PendingRewards:              pendingRewards.Raw,
+		AmountWithoutPendingRewards: amountWithoutPendingRewards.Raw,
+		Rewards:                     record.RewardedMicroAlgos.Raw,
+		Status:                      record.Status.String(),
+		RewardBase:                  &record.RewardsBase,
+		Participation:               apiParticipation,
+		CreatedAssets:               &createdAssets,
+		Assets:                      &assets,
+	}
+
+	return ctx.JSON(http.StatusOK, response)
+}
+
+// GetBlock gets the block for the given round.
+// (GET /v2/blocks/{round})
+func (v2 *Handlers) GetBlock(ctx echo.Context, round uint64, params generated.GetBlockParams) error {
+	handle, err := getCodecHandle(params.Format)
+	if err != nil {
+		return badRequest(ctx, err, errFailedParsingFormatOption, v2.Log)
+	}
+
+	// msgpack format will return the raw block bytes and attach some custom headers.
+	if handle == protocol.CodecHandle {
+		blockbytes, err := rpcs.RawBlockBytes(v2.Node.Ledger(), basics.Round(round))
+		if err != nil {
+			return internalError(ctx, err, errFailedToLookupRawBlock, v2.Log)
+		}
+
+		ctx.Response().Writer.Header().Add("Content-Encoding", "application/x-algorand-block-v1")
```

Comment:
nonstandard use of Content-Encoding. I think we should drop this and just use the X- header
https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.11